src/types/redis_zset.cc

```diff
@@ -224,8 +222,6 @@ rocksdb::Status ZSet::RangeByRank(engine::Context &ctx, const Slice &user_key, c
 
   std::string ns_key = AppendNamespacePrefix(user_key);
 
-  std::optional<LockGuard> lock_guard;
-  if (spec.with_deletion) lock_guard.emplace(storage_->GetLockManager(), ns_key);
```

Comment:
I'm afraid this change the syntax a bit but this lgtm now. Maybe we can losen the check with `spec`

cc @PragmaTwice

Reply:
https://github.com/apache/kvrocks/issues/2563#issuecomment-2423583948

Seems fine.

Reply:
Ok, this is ok to me